### PR TITLE
Remove unnecessary code from RestClientRecorder

### LINKED ISSUE #12325: Remove unnecessary code in RestClientRecorder wrt ResteasyProviderFactory
As of https://issues.redhat.com/browse/RESTEASY-2684 "Set providerFactory on ResteasyClientBuilderImpl", the code in io.quarkus.restclient.runtime.RestClientRecorder which configures ResteasyProviderFactory.instance is no longer necessary.

### PR DESCRIPTION
**Pull request by ronsigal**
Fixes #12325

### CODE REVIEW

File: extensions/rest-client/runtime/src/main/java/io/quarkus/restclient/runtime/RestClientRecorder.java [hunk 1]
```diff
@@ -60,14 +60,6 @@ public InjectorFactory getInjectorFactory() {
         };
 
         registerProviders(clientProviderFactory, useBuiltIn, providersToRegister, contributedProviders);
-
```
Comment (kenfinnigan) [inline extensions/rest-client/runtime/src/main/java/io/quarkus/restclient/runtime/RestClientRecorder.java:63]: I think this is missing the `ResteasyClientBuilderImpl.setProviderFactory(clientProviderFactory)`, similar to the MP REST Client setting

Comment (ronsigal) [reply extensions/rest-client/runtime/src/main/java/io/quarkus/restclient/runtime/RestClientRecorder.java:63]: @kenfinnigan, I'm not following. Line 63 is
```
RestClientBuilderImpl.setProviderFactory(clientProviderFactory);
```

Comment (kenfinnigan) [reply extensions/rest-client/runtime/src/main/java/io/quarkus/restclient/runtime/RestClientRecorder.java:63]: Right, which is the MP REST Client builder impl.

We need to do the same for the JAX-RS Client builder impl, which is `ResteasyClientBuilderImpl`.

The names are close, which makes it confusing

Comment (ronsigal) [reply extensions/rest-client/runtime/src/main/java/io/quarkus/restclient/runtime/RestClientRecorder.java:63]: D'oh!

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**